### PR TITLE
Percona Server 5.6.36-82.0 release notes (5.6)

### PR DESCRIPTION
**Pull request by hrvojem**

### CODE REVIEW

**Review by george-lorch — APPROVED** [review]
Toku +1

File: doc/source/performance/prefix_index_queries_optimization.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.. _prefix_index_queries_optimization:
+
+=================================
+Prefix Index Queries Optimization
+=================================
+
+|Percona Server| has ported Prefix Index Queries Optimization feature from
+Facebook.
```
Comment (laurynas-biveinis) [inline doc/source/performance/prefix_index_queries_optimization.rst]: Facebook patch for MySQL

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+============================
+|Percona Server| 5.6.36-82.0
+============================
+
+Percona is glad to announce the release of |Percona Server| 5.6.36-82.0 on
+May 11, 2017 (Downloads are available `here
+<http://www.percona.com/downloads/Percona-Server-5.6/Percona-Server-5.6.36-82.0/>`_
+and from the :doc:`Percona Software Repositories </installation>`).
+
+Based on `MySQL 5.6.34
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: Based on MySQL 5.6.36

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+ |Percona Server| now supports :ref:`prefix_index_queries_optimization`.
+
+ :variable:`tokudb_dir_cmd` can now be used to :ref:`edit the TokuDB
+ <editing_tokudb_files_with_tokudb_dir_cmd>` files.
+
+Bugs Fixed
+==========
+
+ |Percona Server| would hang during OLTP prepare with several buffer pool
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: What is "OLTP prepare"?

Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: with several small buffer pool instances

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+ Querying :table:`TABLE_STATISTICS` in combination with a stored function could
+ lead to a server crash. Bug fixed :bug:`1659992`.
+
+ :file:`tokubackup_slave_info` file was created for a master server after
+ taking the backup with :ref:`toku_backup`. Bug fixed :bkpbug:`135`.
+
+ Fixed a memory leak in :ref:`toku_backup`. Bug fixed :bug:`1669005`.
+
+ :ref:`compressed_columns` could not be added to a partitioned table by using
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst:47]: How does the output text look here?

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ ``ALTER TABLE``. Bug fixed :bug:`1671492`.
+
+ Fixed a memory leak which happened that happened in case of failure to create
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: which happened that happened

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ a multi-threaded slave worker thread. Bug fixed :bug:`1675716`.
+
+ Combination of using the :ref:`audit_log_plugin` with MB collation connection
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: Not audit_log_plugin but any audit API-using plugin (audit_log, QRT, ...)

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ and ``PREPARE`` statement with a parse error could lead to a server crash. Bug
+ fixed :bug:`1688698` (upstream :mysqlbug:`86209`).
+
+ Fix for a :bug:`1433432` bug in |Percona Server| rn:`5.6.28-76.1` caused a
+ performance regression. Bug fixed :bug:`1631309`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: regression due to suboptimal LRU manager thread flushing heuristics

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+ Creating :ref:`compressed_columns` in |MyISAM| tables by specifying partition
+ engines would not result in error. Bug fixed :bug:`1631954`.
+
+ It was not possible to configure basedir as a symlink. Bug fixed
+ :bug:`1639735`.
+
+ Slave did not report a ``Seconds_Behind_Master`` with multiple number of slave
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: Replication slave did not report SBM correctly when running in multi-threaded slave mode

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by laurynas-biveinis** [comment]
Please also amend the commit message for the correct version number

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+ |Percona Server| now supports :ref:`prefix_index_queries_optimization`.
+
+ :variable:`tokudb_dir_cmd` can now be used to :ref:`edit the TokuDB
+ <editing_tokudb_files_with_tokudb_dir_cmd>` files.
+
+Bugs Fixed
+==========
+
+ Deadlock could occur when server was using several small buffer pool instances
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: Deadlock could occur in I/O-bound workloads when ...

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst (hunk 4 above)
Comment (laurynas-biveinis) [reply doc/source/release-notes/Percona-Server-5.6.36-82.0.rst:47]: OK as discussed on Slack

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+ Fixed a memory leak in :ref:`toku_backup`. Bug fixed :bug:`1669005`.
+
+ :ref:`compressed_columns` could not be added to a partitioned table by using
+ ``ALTER TABLE``. Bug fixed :bug:`1671492`.
+
+ Fixed a memory leak that happened in case of failure to create
+ a multi-threaded slave worker thread. Bug fixed :bug:`1675716`.
+
+ Combination of using any audit API-using plugin (audit_log, QRT, ...) with MB
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: As discussed on Slack

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: doc/source/release-notes/Percona-Server-5.6.36-82.0.rst [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+ |Percona Server| now supports :ref:`prefix_index_queries_optimization`.
+
+ :variable:`tokudb_dir_cmd` can now be used to :ref:`edit the TokuDB
+ <editing_tokudb_files_with_tokudb_dir_cmd>` files.
+
+Bugs Fixed
+==========
+
+ Deadlock could occur in I/O-bound workloads when server was using several
+ small buffer pool instances in combination with
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.6.36-82.0.rst]: several buffer pool instances in combination with small redo log files and variable ...

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**